tools/bpf2c/bpf_code_generator.cpp

```diff
@@ -246,18 +255,31 @@ bpf_code_generator::bpf_code_generator(
     const bpf_code_generator::unsafe_string& c_name, const std::vector<ebpf_inst>& instructions)
     : c_name(c_name)
 {
-    current_program = &programs[c_name];
-    get_register_name(0);
-    get_register_name(1);
-    get_register_name(10);
+    bpf_code_generator_program* current_program = add_program(c_name, c_name);
     uint32_t offset = 0;
     for (const auto& instruction : instructions) {
-        current_program->output.push_back({instruction, offset++});
+        std::string unsafe_name = "local_subprogram" + std::to_string(offset);
```

Comment:
Do we care about this colliding with a user provided name at all? I suspect not, but something to consider / maybe document. Generating symbols that may collide with user generated symbols is a small worry I have had in the past.

Reply:
Added a comment to this effect.